platform/security/sts/security-sts-ldaplogin/src/test/java/ddf/ldap/ldaplogin/LdapModuleTest.java

```diff
@@ -0,0 +1,208 @@
+        options.put(ROLE_FILTER, "(member=uid=%u,ou=users,dc=example,dc=com)");
+        options.put(ROLE_BASE_DN, "ou=groups,dc=example,dc=com");
+        options.put(ROLE_NAME_ATTRIBUTE, "cn");
+        options.put(ROLE_SEARCH_SUBTREE, "true");
+        options.put(SSL_STARTTLS, "false");
+        options.put(BIND_METHOD, "Simple");
+        options.put(REALM, "");
+        options.put(KDC_ADDRESS, "");
+
+        // If the class-under-test cannot get the encryption service,
+        // assume the password is already decrypted.
+        options.put(CONNECTION_PASSWORD, "secret");
+    }
+
+    @After
+    public void shutdown() {
+        if (server != null) {
+            server.stop();
+        }
+    }
+
+    @Test
+    public void testSuccessfulLdapLogin() throws InterruptedException, LoginException {
+
+        CallbackHandler mockHandler = getNamePasswordHandler("tstark", "password1");
+        SslLdapLoginModule module = getTestModule();
+        module.initialize(new Subject(), mockHandler, new HashMap<String, String>(), options);
+        module.login();
+
+        Set<Principal> principals;
+        principals = module.getPrincipals();
+        assertThat(principals,
+                containsInAnyOrder(new UserPrincipal("tstark"), new RolePrincipal("avengers")));
+    }
+
+    @Test
+    public void testIncorrectPassword() throws LoginException {
+        CallbackHandler mockHandler = getNamePasswordHandler("tstark",
+                "Ce n'est pas un mot de passe");
+        SslLdapLoginModule module = getTestModule();
+        module.initialize(new Subject(), mockHandler, new HashMap<String, String>(), options);
+        module.login();
+        assertThat("Expected no principals assigned to user", module.getPrincipals(), empty());
+    }
+
+    @Test(expected = LoginException.class)
+    public void testInvalidPassword() throws LoginException {
+        CallbackHandler mockHandler = getNamePasswordHandler("<>", "");
+        SslLdapLoginModule module = getTestModule();
+        module.initialize(new Subject(), mockHandler, new HashMap<String, String>(), options);
+        module.login();
+    }
+
+    @Test
+    public void testLdapsLogin() {
+        // TODO: DDF-2876 - Enhance LDAP tests to use SSL/TLS
+    }
+
+    @Test
+    public void testStartTls() {
+        // TODO: DDF-2876 - Enhance LDAP tests to use SSL/TLS
+    }
+
+    @Test
+    public void testSasl() {
+        // TODO: DDF-2877 - Enhance LDAP test for alt protocols
+    }
+
+    @Test
+    public void testGssapiSasl() {
+        // TODO: DDF-2877 - Enhance LDAP test for alt protocols
+
+    }
+
+    @Test
+    public void testDigestMd5Sasl() {
+        // TODO: DDF-2877 - Enhance LDAP test for alt protocols
+
+    }
+
+    private SslLdapLoginModule getTestModule() {
+        SslLdapLoginModule module = new SslLdapLoginModule();
+        module.setEncryptionService(new EncryptionServiceImpl());
+        return module;
+    }
```

Comment:
Refactored tested-class and the test to inject the encryption service. The test no longer creates an anonymous class that overrides the getDecryptedPassword() method. @pklinef